### PR TITLE
refactor: speed up IPFS resource loading

### PR DESCRIPTION
**Pull request by UncleBill**

### CODE REVIEW

File: packages/web3-shared/base/tests/resolver.ts [hunk 1]
```diff
@@ -27,14 +27,17 @@ describe('resolveIPFSLink', () => {
...
             { give: '', expected: '' },
-            { give: cid, expected: `https://ipfs.io/ipfs/${cid}` },
-            { give: `${cid}/image`, expected: `https://ipfs.io/ipfs/${cid}/image` },
-            { give: `ipfs://${cid}`, expected: `https://ipfs.io/ipfs/${cid}` },
-            { give: `ipfs://ipfs/${cid}`, expected: `https://ipfs.io/ipfs/${cid}` },
-            { give: `https://ipfs.io/ipfs/${IMAGE_DATA_PARTIAL}`, expected: IMAGE_DATA_PARTIAL },
-            { give: `https://ipfs.io/ipfs/${cid}`, expected: `https://ipfs.io/ipfs/${cid}` },
-            { give: `https://ipfs.altava.com/ipfs/${cid}`, expected: `https://ipfs.io/ipfs/${cid}` },
-            { give: `https://ipfs.altava.com/ipfs/${cid}/image`, expected: `https://ipfs.io/ipfs/${cid}/image` },
+            { give: cid, expected: `https://gateway.ipfscdn.io/ipfs/${cid}` },
```
Comment (guanbinrui) [inline packages/web3-shared/base/tests/resolver.ts:30]: Add cases for trimming query part.

**Review by Jack-Works — CHANGES_REQUESTED** [review]
we can have a try on this, but please fallback to current behavior on iOS. `process.env.arch` and `process.env.engine`

File: packages/mask/src/manifest-v3.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "description": "The portal to the new & open Internet. Send encrypted message and decentralized Apps right on top of social networks.",
     "web_accessible_resources": [
         {
-            "resources": ["js/*", "*.svg", "*.png", "*.css"],
+            "resources": ["js/*", "*.svg", "*.png", "*.css", "https://gateway.ipfscdn.io/ipfs/*"],
```
Comment (guanbinrui) [inline packages/mask/src/manifest-v3.json:16]: I'm afraid this change causes users will fail to automatic upgrading the extension.

**Comment by UncleBill** [comment]
> we can have a try on this, but please fallback to current behavior on iOS. `process.env.arch` and `process.env.engine`

We access web content by https://github.com/DimensionDev/Maskbook/blob/ef23fbff0ae3eb1e1bbac0bee8384110fd2fab19/packages/shared/src/UI/components/Image/loadImage.ts#L6-L30

We can easily bypass it for iOS and Firefox, and fallback to r2d2Fetch which fetches via background.

File: packages/web3-shared/base/tests/resolver.ts [hunk 3]
```diff
@@ -27,15 +27,22 @@ describe('resolveIPFSLink', () => {
...
+            { give: `https://gateway.ipfscdn.io/ipfs/${cid}`, expected: `https://gateway.ipfscdn.io/ipfs/${cid}` },
+            { give: `https://ipfs.altava.com/ipfs/${cid}`, expected: `https://gateway.ipfscdn.io/ipfs/${cid}` },
+            {
+                give: `https://ipfs.altava.com/ipfs/${cid}/image`,
+                expected: `https://gateway.ipfscdn.io/ipfs/${cid}/image`,
+            },
             { give: MASK_URL, expected: MASK_URL },
+            {
+                give: 'https://gateway.ipfscdn.io/ipfs/bafybeig2k4aoikypyfov6phu47linn2ygra5oilyhvyw6slxsvw6iafhui?id=67874',
+                expected: 'https://gateway.ipfscdn.io/ipfs/bafybeig2k4aoikypyfov6phu47linn2ygra5oilyhvyw6slxsvw6iafhui',
```
Comment (guanbinrui) [inline packages/web3-shared/base/tests/resolver.ts]:
```suggestion
                give: 'https://gateway.ipfscdn.io/ipfs/${cid}?id=67874',
                expected: 'https://gateway.ipfscdn.io/ipfs/${cid}',
```

File: packages/mask/src/manifest-v3.json (hunk 2 above)
Comment (UncleBill) [reply packages/mask/src/manifest-v3.json:16]: We already did once when we refactored our Icons

File: packages/web3-shared/base/tests/resolver.ts (hunk 3 above)
Comment (UncleBill) [reply packages/web3-shared/base/tests/resolver.ts]: done

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**